src/lib/validation/documentation.ts

```diff
@@ -0,0 +1,38 @@
+import * as path from "path";
+import * as ts from "typescript";
+import { ProjectReflection, ReflectionKind } from "../models";
+import { Logger, normalizePath } from "../utils";
+
+export function validateDocumentation(
+    project: ProjectReflection,
+    logger: Logger,
+    requiredToBeDocumented: readonly (keyof typeof ReflectionKind)[]
+): void {
+    const kinds = requiredToBeDocumented.reduce(
+        (prev, cur) => (prev |= ReflectionKind[cur]),
+        0
+    );
+
+    for (const ref of project.getReflectionsByKind(kinds)) {
+        const symbol = project.getSymbolFromReflection(ref);
+        if (!ref.comment && symbol?.declarations) {
+            const decl = symbol.declarations[0];
+            const sourceFile = decl.getSourceFile();
+            const { line } = ts.getLineAndCharacterOfPosition(
+                sourceFile,
+                decl.getStart()
+            );
+            const file = normalizePath(
+                path.relative(process.cwd(), sourceFile.fileName)
+            );
+
+            if (file.startsWith(`node_modules${path.sep}`)) {
```

Comment:
I think this will break badly in monorepos where node_modules is in a directory above where typedoc might be run.

Reply:
How about an option to specify folders to ignore within?

Reply:
How would you recommend going about this? For the usecase that I have I don't want to block building docs if a dependency doesn't have docs.

Reply:
I think just `file.includes("node_modules")` would be fine. In a parent directory we'd end up with `"../node_modules/foo"`, which we shouldn't warn about.

Reply:
Sounds good